components/builder-header/index.jsx

```diff
@@ -0,0 +1,200 @@
+	 * * `onClickHelp`: Called when the Help link is clicked.
+	 * _Tested with Mocha testing._
+	 */
+	events: PropTypes.shape({
+		onClickBack: PropTypes.func,
+		onClickHelp: PropTypes.func,
+	}),
+	/**
+	 * **Text labels for internationalization**
+	 * This object is merged with the default props object on every render.
+	 * * `back`: The label for the Back link.
+	 * * `help`: The label for the Help link.
+	 * * `pageType`: The label that describes the page type.
+	 * * `title`: The label for the page title.
+	 * _Tested with snapshot testing._
+	 */
+	labels: PropTypes.shape({
+		back: PropTypes.string,
+		help: PropTypes.string,
+		pageType: PropTypes.string,
+		title: PropTypes.string,
+	}),
+	/**
+	 * Custom styles to be passed to the component
+	 */
+	style: PropTypes.object,
+};
+
+const defaultProps = {
+	assistiveText: {
+		backIcon: 'Back',
+		helpIcon: 'Help',
+		icon: 'Builder',
+	},
+	labels: {
+		back: 'Back',
+		help: 'Help',
+		pageType: 'Page Type',
+		title: 'App Name',
+	},
+};
+
+/**
+ * Every builder needs a builder header, which contains basic navigation elements. It also shows the builder type and content name.
+ */
+const BuilderHeader = (props) => {
+	const assistiveText = {
+		...defaultProps.assistiveText,
+		...props.assistiveText,
+	};
+	const events = {
+		...{},
+		...props.events,
+	};
+	const labels = {
+		...defaultProps.labels,
+		...props.labels,
+	};
+
+	let nav;
+	let toolbar;
+	React.Children.forEach(props.children, (child) => {
+		if (child) {
+			switch (child.type.displayName) {
+				case BUILDER_HEADER_NAV:
+					nav = child;
+					break;
+				case BUILDER_HEADER_TOOLBAR:
+					toolbar = child;
+					break;
+				default:
+			}
+		}
+	});
+
+	return (
+		<div style={{ position: 'relative', height: '100px' }}>
+			<div
+				className={classNames('slds-builder-header_container', props.className)}
+				style={props.style}
+			>
+				<header className="slds-builder-header">
+					<div className="slds-builder-header__item">
+						<div className="slds-builder-header__item-label slds-media slds-media_center">
+							<div className="slds-media__figure">
+								<Icon
+									assistiveText={{ label: assistiveText.icon }}
+									category="utility"
+									containerClassName="slds-icon_container slds-icon-utility-builder slds-current-color"
+									name="builder"
+									size="x-small"
+								/>
+							</div>
+							<div className="slds-media__body">{labels.title}</div>
+						</div>
+					</div>
+					{nav}
+					<div className="slds-builder-header__item slds-has-flexi-truncate">
+						<h1 className="slds-builder-header__item-label">
+							<span className="slds-truncate" title={labels.pageType}>
+								{labels.pageType}
+							</span>
+						</h1>
+					</div>
+					<div className="slds-builder-header__item slds-builder-header__utilities">
+						<div className="slds-builder-header__utilities-item">
+							<a
+								href="javascript:void(0);"
```

Comment:
I wasn't sure if we wanted to pass in `<a>` attributes like href, target, etc. to use with links instead of the `onClick`.  I think it's ok to just have `onClick`?

Reply:
onClick is always best I think, since you can do a location change, but not the reverse.